test/zdtm.py

```diff
@@ -1287,7 +1299,15 @@ def spawn_criu_image_streamer(self, action):
                 raise test_fail_exc(
                     "criu-image-streamer is not starting (exit_code=%d)" % p.wait())
 
-        return p
+        progress.close()
+
+        self.__img_streamer_process = p
+
+    def wait_for_criu_image_streamer(self):
+        ret = self.__img_streamer_process.wait()
```

Comment:
It might be good to set a `timeout` value for `wait()`.

Reply:
Thanks, I will try to come up with a more general solution for this problem. Also, I noticed a few places where `wait()` is used with stdout/stderr set to `PIPE`, and it looks like this may cause a deadlock: https://docs.python.org/3.8/library/subprocess.html#subprocess.Popen.wait

Reply:
Regarding deadlocks, I fixed some of them here: https://github.com/checkpoint-restore/criu/pull/1642 but I didn't think this PR was worth looking at, but it is. Let me re-open it.